packages/@aws-cdk/aws-appmesh/lib/tls-validation-context.ts

```diff
@@ -0,0 +1,160 @@
+import * as acmpca from '@aws-cdk/aws-acmpca';
+import { CfnVirtualGateway, CfnVirtualNode } from './appmesh.generated';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct } from '@aws-cdk/core';
+
+/**
+ * Represents the properties needed to define TLS validation context
+ */
+export interface TlsValidationContext {
+  /**
+   * Reference to where to retrieve the trust chain.
+   */
+  readonly trust: TlsValidationContextTrust;
+}
+
+/**
+ * All Properties for TLS Validations for both Client Policy and Listener.
+ */
+export interface TlsValidationTrustConfig {
+  /**
+   * VirtualNode CFN configuration for client policy's TLS Validation
+   */
+  readonly virtualNodeClientTlsValidationContextTrust: CfnVirtualNode.TlsValidationContextTrustProperty;
+
+  /**
+   * VirtualNode CFN configuration for listener's TLS Validation
+   *
+   * @default - no TLS Validation
+   */
+  readonly virtualNodeListenerTlsValidationContextTrust?: CfnVirtualNode.ListenerTlsValidationContextTrustProperty
+
+  /**
+   * VirtualGateway CFN configuration for client policy's TLS Validation
+   */
+  readonly virtualGatewayClientTlsValidationContextTrust: CfnVirtualGateway.VirtualGatewayTlsValidationContextTrustProperty;
+
+  /**
+   * VirtualGateway CFN configuration for listener's TLS Validation
+   *
+   * @default - no TLS Validation
+   */
+  readonly virtualGatewayListenerTlsValidationContextTrust?: CfnVirtualGateway.VirtualGatewayListenerTlsValidationContextTrustProperty;
+}
+
+/**
+ * ACM Trust Properties
+ */
+export interface AcmTrustOptions {
+  /**
+   * Contains information for your private certificate authority
+   */
+  readonly certificateAuthorities: acmpca.ICertificateAuthority[];
+}
+
+/**
+ * File Trust Properties
+ */
+export interface FileTrustOptions {
+  /**
+   * Path to the Certificate Chain file on the file system where the Envoy is deployed.
+   */
+  readonly certificateChain: string;
+}
+
+/**
+ * Defines the TLS validation context trust.
+ */
+export abstract class TlsValidationContextTrust {
+  /**
+   * Tells envoy where to fetch the validation context from
+   */
+  public static fileTrust(props: FileTrustOptions): TlsValidationContextTrust {
```

Comment:
Now that "trust" is in the name, I would rename this to just `file`:

```suggestion
  public static file(props: TlsValidationFileTrustOptions): TlsValidationTrust {
```

Reply:
Good point! Addressed.